gateway/proxy_muxer.go

```diff
@@ -287,7 +287,11 @@ func (gw *Gateway) flushNetworkAnalytics(ctx context.Context) {
 					APIID:        spec.APIID,
 					OrgID:        spec.OrgID,
 				}
-				record.SetExpiry(spec.ExpireAnalyticsAfter)
+				if spec.DisableExpireAnalytics {
+					record.SetExpiry(0)
+				} else {
+					record.SetExpiry(spec.ExpireAnalyticsAfter)
+				}
```

Comment:
```suggestion
				expiresAfter := spec.ExpireAnalyticsAfter
				if spec.DisableExpireAnalytics {
					expiresAfter = 0
				}
				record.SetExpiry(expiresAfter)
```

This tries to match the handlers expiry patterns.